doc/tutorials/distributed.rst

```diff
@@ -4,6 +4,105 @@ Distributed Secure XGBoost
+
+The tracker is logically distinct from all nodes in the cluster, and does not actually perform any computation. Once it has finished setting up the topology, the tracker has finished its job. However, a user may choose to colocate the tracker with a master/worker node for simplicity. 
+
+Before training begins, Distributed Secure XGBoost leverages *inter-enclave attestation* to authenticate all enclaves in the cluster -- each enclave attests all neighboring enclaves to verify the enclave's identity and to verify that the enclave will be running the proper code.
+   
+
+********************
+Distributed Training
+********************
+
+In the distributed setting, distributed training occurs by partitioning a subset of the training data to each node -- each node is responsible for some chunk of the training data when computing the global model. 
+
+During training, the nodes compute a global model by iteratively communicating optimal feature splits to neighboring nodes. These splits are aggregated at the master node through the Allreduce paradigm, enabling the computation of a globally optimal feature split per iteration. This exchange of local optimal feature splits gradually builds up a tree, and a tree ensemble is built up with gradient boosting to form a model. 
+
+
+.. image:: images/step1.png
+   :width: 100% 
+   :alt: Step 1
+
+.. image:: images/step2.png
+   :width: 100% 
+   :alt: Step 2 
+
+.. image:: images/step3.png
+   :width: 100% 
+   :alt: Step 3
+
+.. image:: images/step4.png
+   :width: 100% 
+   :alt: Step 4 
+
+For more information on the distributed training algorithm, see section 3 of the `original XGBoost paper <https://arxiv.org/pdf/1603.02754.pdf>`_. 
+
+********
+Tutorial
+********
+This tutorial demonstrates how to run Distributed Secure XGBoost. The tutorial code is located at ``demo/python/distributed/``.
+
+If you are using Azure Confidential Computing, all your VMs must be on the same subnet.
+
+1. Modify ``demo/python/distributed/hosts.config`` to contain the IP addresses of the nodes in your cluster. For example, if the nodes in your cluster have IP addresses of ``13.95.157.223`` and ``40.68.135.193``, your ``hosts.config`` should look like the following.
+
+   .. code-block:: none
+      
+      13.95.157.223:22
+      40.68.135.193:22
+
+2. Ensure that ``distr-training.py`` is identical and at the same path on every machine in the cluster. 
+
+3. Start computation. This will start the tracker and subsequently the job.
+
+   .. code-block:: bash
+
+      mc2-xgboost/host/dmlc-core/tracker/dmlc-submit --cluster ssh --host-file hosts.config --num-workers <num_nodes_in_cluster> --worker-memory 4g python3 distr-training.py
+
+
+   Ensure that each node in the cluster has authorized the SSH public key of the machine running the tracker, as the tracker in Distributed Secure XGBoost leverages SSH public keys to set up the topology.
```

Comment:
I think we should introduce the SSH-based connections earlier, perhaps in the architecture...

Reply:
Fixed